### PR TITLE
Deploy to Kaia

### PR DESCRIPTION
**Pull request by ezynda3**
# Which Jira task belongs to this PR?

# Why did I implement it this way?

# Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] This pull request is as small as possible and only tackles one problem
- [ ] I have added tests that cover the functionality / test the bug
- [ ] I have updated any required documentation

# Checklist for reviewer (DO NOT DEPLOY and contracts BEFORE CHECKING THIS!!!)

- [ ] I have checked that any arbitrary calls to external contracts are validated and or restricted
- [ ] I have checked that any privileged calls (i.e. storage modifications) are validated and or restricted
- [ ] I have ensured that any new contracts have had AT A MINIMUM 1 preliminary audit conducted on <date> by <company/auditor>

### CODE REVIEW

File: deployments/_deployments_log_file.json [hunk 1]
```diff
@@ -669,6 +669,20 @@
...
+    "kaia": {
+      "production": {
+        "1.0.0": [
+          {
+            "ADDRESS": "0x8176C5502f1A01fCeB8d14fdc7ad15867ca562C1",
+            "OPTIMIZER_RUNS": "1000000",
+            "TIMESTAMP": "2024-09-05 16:24:10",
+            "CONSTRUCTOR_ARGS": "0x",
+            "SALT": "",
+            "VERIFIED": "true"
```
Comment (0xDEnYO) [inline deployments/_deployments_log_file.json:682]: https://kaiascan.io/address/0x8176c5502f1a01fceb8d14fdc7ad15867ca562c1?tabId=contract&page=1

This does not seem verified to me

File: networks [hunk 2]
```diff
@@ -13,6 +13,7 @@ fuse
 gnosis
 gravity
 immutablezkevm
+kaia
```
Comment (0xDEnYO) [inline networks:16]: please could you also update config/networks.json (while they are still running in parallel)?

File: script/deploy/healthCheck.mts [hunk 3]
```diff
@@ -109,7 +109,7 @@ const main = defineCommand({
       `../../script/deploy/_targetState.json`
     )
     const nonCoreFacets = Object.keys(
-      targetStateJson[network.toLowerCase()].production.LiFiDiamond
+      targetStateJson[network.toLowerCase()].production.LiFiDiamond,
```
Comment (0xDEnYO) [inline script/deploy/healthCheck.mts:112]: why is it adding a , at the end?

File: script/deploy/healthCheck.mts [hunk 4]
```diff
@@ -178,7 +178,7 @@ const main = defineCommand({
       const isDeployed = await checkIsDeployed(
         facet,
         deployedContracts,
-        publicClient
+        publicClient,
```
Comment (0xDEnYO) [inline script/deploy/healthCheck.mts:181]: same here...publicClient is the last parameter, there shouldnt be a comma at the end imho

File: script/deploy/safe/config.ts [hunk 5]
```diff
@@ -49,6 +50,7 @@ export const safeAddresses: Record<string, string> = {
   gnosis: '0x2bC523875b59A1Ddd03CEB1F1b28c5B0e8e6654A',
   gravity: '0x245B16CaCE8730b009c5352186DcE7d73c3037A1',
   immutablezkevm: '0xa89a87986e8ee1Ac8fDaCc5Ac91627010Ec9f772',
+  kaia: '0xa89a87986e8ee1Ac8fDaCc5Ac91627010Ec9f772',
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply script/deploy/safe/config.ts:53]: @ezynda3 please respond to such comments from coderabbit and explain your reasoning so we can train the AI

File: script/helperFunctions.sh [hunk 6]
```diff
@@ -3858,4 +3858,4 @@ function test_tmp() {
   # RESPONSE=$(yarn add-safe-owners --network taiko --rpc-url $(getRPCUrl $NETWORK $ENVIRONMENT) --privateKey $PRIVATE_KEY_PRODUCTION --owners "0xb78FbE12d9C09d98ce7271Fa089c2fe437B7B4D5,0x65f6F29D3eb871254d71A79CC4F74dB3AAF3b86e,0x24767E3A1cb07ee500BA9A5621F2B608440Ca270,0x81Dbb716aA13869323974A1766120D0854188e3e,0x11F1022cA6AdEF6400e5677528a80d49a069C00c,0x498E8fF83B503aDe5e905719D27b2f11B605b45A")
   # echo "RESPONSE: $RESPONSE"
 }
-test_tmp
+#test_tmp
```
Comment (0xDEnYO) [inline script/helperFunctions.sh:3861]: if you merge the latest `main` this should be removed

**Review by 0xDEnYO — CHANGES_REQUESTED** [review]

File: script/deploy/safe/config.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (ezynda3) [reply script/deploy/safe/config.ts:53]: This is fine as contracts can have the same address on different networks.

File: script/deploy/healthCheck.mts (hunk 3 above)
Comment (ezynda3) [reply script/deploy/healthCheck.mts:112]: This is pretty common. It's mainly used to be able to add new items later without having to remember to add another comma in the line above.

File: deployments/_deployments_log_file.json (hunk 1 above)
Comment (ezynda3) [reply deployments/_deployments_log_file.json:682]: This was the official explorer as far as I was aware https://kaiascope.com/account/0x8176C5502f1A01fCeB8d14fdc7ad15867ca562C1?tabId=internalTx

File: deployments/kaia.diamond.json
Comment (0xDEnYO) [inline deployments/kaia.diamond.json:1]: what happened here?

**Review by 0xDEnYO — APPROVED** [review]

File: deployments/_deployments_log_file.json (hunk 1 above)
Comment (0xDEnYO) [reply deployments/_deployments_log_file.json:682]: May be....I found this other one linked on the homepage of Kaia chain (bottom right):
https://www.kaia.io/

But OK, as long as we have them verified in one explorer we can point to that...just thought the verification didnt work at all

**Review by 0xDEnYO — APPROVED** [review]

**Pull request merged**